tests/src/rules/no-useless-path-segments.js

```diff
@@ -0,0 +1,50 @@
+import { test } from '../utils'
+import { RuleTester } from 'eslint'
+
+const ruleTester = new RuleTester()
+const rule = require('rules/no-useless-path-segments')
+
+function runResolverTests(resolver) {
+  ruleTester.run(`no-useless-path-segments (${resolver})`, rule, {
+    valid: [
+      test({ code: 'import "./malformed.js"' }),
+      test({ code: 'import "./test-module"' }),
+      test({ code: 'import "."' }),
+      test({ code: 'import ".."' }),
+      test({ code: 'import fs from "fs"' }),
+    ],
+
+    invalid: [
+      test({
+        code: 'import "./../files/malformed.js"',
+        errors: [ 'Useless path segments for "./../files/malformed.js", should be "../files/malformed.js"'],
+      }),
+      test({
+        code: 'import "./../files/malformed"',
+        errors: [ 'Useless path segments for "./../files/malformed", should be "../files/malformed"'],
+      }),
+      test({
+        code: 'import "../files/malformed.js"',
+        errors: [ 'Useless path segments for "../files/malformed.js", should be "./malformed.js"'],
+      }),
+      test({
+        code: 'import "../files/malformed"',
+        errors: [ 'Useless path segments for "../files/malformed", should be "./malformed"'],
+      }),
+      test({
+        code: 'import "./test-module/"',
```

Comment:
We should test all of these cases:
1. `foo/index.something` exists → path should be "foo"
1. `foo.something` exists → path should be "foo"
1. both `foo/index.something` and `foo.something` exist → `foo/` and `foo` are different, and the difference is important, and this rule should not warn on it.